pkg/commands/cmdlib/global_options.go

```diff
@@ -42,6 +42,10 @@ type GlobalOptions struct {
 	// InlineObjects indicates whether to inline the objects before doing any
 	// processing (typically requires InlineComponents to be useful).
 	InlineObjects bool
+
+	// AllowUnknownFields indicates whether to allow unknown fields during
+	// decoding. Defaults to false.
+	AllowUnknownFields bool
```

Comment:
That's a really good question. Go defaults to allowing unknown fields. I'm not sure why that was chosen. But it seems like people run into this case frequently: https://github.com/ghodss/yaml/commit/8aa042675dd6a7951644be0acb134430ed0dd58b

I suppose it might be useful as a hacky backwards compatibility method, albeit one that's likely to cause bugs. Maybe I'll remove from the CLI.

Reply:
I removed it as an option from the CLI

Reply:
This is great.  The less the merrier!